File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int size = height.size();
        int left = 0;
        int maxArea = 0;
        int right = size - 1;
        
        while (left < right) {
            int dis = right - left;
            int area = min(height[left], height[right]) * dis;
            maxArea = max(maxArea, area);
            
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        
        return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int left=0;
        int right=n-1;
        int min=0;
        int result=0;
        while(left<right){
            min=Math.min(height[left],height[right]);
            result=Math.max(result, min*(right-left));
            if(height[left]<height[right]){
                left++;
            }else{
                right--;
            }
        }
        return result;
    }
}